workers/data_refinery_workers/downloaders/transcriptome_index.py

```diff
@@ -82,61 +84,68 @@ def download_transcriptome(job_id: int) -> None:
-        success = False
 
-    if success:
-        try:
-            _verify_files(first_fasta_file, second_fasta_file, job)
-            _verify_files(first_gtf_file, second_gtf_file, job)
-
-            # The two Batches share the same fasta and gtf files, so
-            # only download each one once
-            _download_file(first_fasta_file.download_url,
-                           first_fasta_file.get_temp_pre_path(job_dir),
-                           job)
-            _download_file(first_gtf_file.download_url,
-                           first_gtf_file.get_temp_pre_path(job_dir),
-                           job)
-
-            # Then create symlinks so the files for the second Batch
-            # can be found where they will be expected to.
-            try:
-                os.symlink(first_fasta_file.get_temp_pre_path(job_dir),
-                           second_fasta_file.get_temp_pre_path(job_dir))
-                os.symlink(first_gtf_file.get_temp_pre_path(job_dir),
-                           second_gtf_file.get_temp_pre_path(job_dir))
-            except Exception:
-                logger.exception("Exception caught while creating symlinks.",
-                                 downloader_job=job.id)
-                job.failure_reason = "Exception caught while creating symlinks."
-                raise
-
-            _upload_files(job_dir,
-                          [first_fasta_file, first_gtf_file, second_fasta_file, second_gtf_file],
-                          job)
-        except Exception:
-            # Exceptions are already logged and handled.
-            # Just need to mark the job as failed.
-            success = False
+    file_assocs = DownloaderJobOriginalFileAssociation.objects.filter(downloader_job=job)
+    files_to_process = []
+    for assoc in file_assocs:
+        original_file = assoc.original_file
+
+        if original_file.is_archive:
+            file_name_species = ''.join(original_file.source_filename.split('.')[:-2])
+        else:
+            # Does this ever happen?
+            file_name_species = ''.join(original_file.source_filename.split('.')[:-1])
+
+        os.makedirs(LOCAL_ROOT_DIR + '/' + file_name_species, exist_ok=True)
+        dl_file_path = LOCAL_ROOT_DIR + '/' + file_name_species + '/' + original_file.source_filename
+        success = _download_file(original_file.source_url, dl_file_path, job)
+
+        if success:
+            original_file.is_downloaded = True
+            original_file.absolute_file_path = dl_file_path
+            original_file.file_name = original_file.source_filename
+            original_file.is_archive = True
+            original_file.has_raw = False
+            original_file.calculate_size()
+            original_file.calculate_sha1()
+            original_file.save()
+            files_to_process.append(original_file)
+        else:
+            logger.error("Problem during download")
```

Comment:
Could you add some additional context to this error statement? I can only imagine looking at a log file and coming across a message like this and wondering "What file, for what downloader job?"

Reply:
👍